apis/core.oam.dev/common/types.go

```diff
@@ -120,7 +119,7 @@ type Terraform struct {
 	// Path is the sub-directory of remote git repository. It's valid when remote is set
 	Path string `json:"path,omitempty"`
 
-	v1beta1.BaseConfigurationSpec `json:",inline"`
+	v1beta2.BaseConfigurationSpec `json:",inline"`
```

Comment:
please make sure the compatibility

Reply:
Yes, the compatibility matter is ensured.
- For an existing v1beta1 Configuration, it won't be affected.
- If a v1beta1 Configuration is updated, it will be automatically converted to v1beta2. 
All its metadata and spec field will be converted in a v1beta2 Configuration as v1beta2 includes all fields of v1beta2. Especially, the `spec.region` field will still be reserved but will treat as a normal variable of Terraform modules. If one wants to set a region to override the one in a Provider, he should use v1beta2 Configuration. For a more detailed background, please refer to this PR comment.